### DISCUSSION TITLE
Delegate Script Settings

### DISCUSSION THREAD

**Discussion by uGhz** [cantaloupe-project/cantaloupe#487] [Q&A] [answered]
Good morning all,

I need to use in my delegate script a path prefix which depends on the server deployment environment. The storage directories for our images cannot be mapped the same way on the development server and on the production server.

What is the best way to make this prefix configurable and avoid having to modify the script itself during deployment? I'm thinking of a specific config file, but I can't figure out the way to access it from the delegate script. Do you have any advice ?

Thanks for your attention !

**Comment by bcail** [comment, marked as answer]
If the config file is in a consistent location relative to your delegate script, you can just open the config file and read it from your ruby (or java) code. Would that work for you?

**Reply by uGhz** [reply]
Yes it would, but I'm so new to Ruby that I can figure out how to open that file: The current directory I can print with "\_\_dir\_\_", for example, is ".../java/javapath" and not the real delegates.rb directory.

I'm probably missing something obvious, right?

**Reply by bcail** [reply]
try looking at this guide: https://www.rubyguides.com/2015/05/working-with-files-ruby/. Using full absolute paths might help.

**Reply by adolski** [reply]
This makes me wonder if it would be helpful to have a VM argument for the delegate script path like the one for the config file: `-Dcantaloupe.delegate=...` (or similar)

**Reply by uGhz** [reply]
Another possibility could be to access, from the script, some custom properties defined in cantaloupe.properties. Is this already possible?

The only context I was able to access was the query context, which doesn't contain a lot of information.

**Reply by adolski** [reply]
@uGhz You could do something like: 

```ruby
# top of script
require 'java'
java_import edu.illinois.library.cantaloupe.config.Configuration

# within a method
Configuration.getInstance.getString('my.custom.config.key')
```

**Reply by uGhz** [reply]
Thank you very much, it works and I really like it!

And you saved my next Monday.

As a reminder, in my delegate script (Cantaloupe version 4.1), the following variant worked:

`custom_prop = Java::edu.illinois.library.cantaloupe.config.Configuration.getInstance.getString('my_custom.prop_key')`

Have an excellent and fulfilling day !
